internal/testing/factory/factory.go

```diff
@@ -71,6 +72,12 @@ func (f *Factory) WithApplicationService(s *applicationservice.WatchableService)
 	return f
 }
 
+// WithModelConfigService configures the factory to use the specified model config service.
+func (f *Factory) WithModelConfigService(s state.ModelConfigService) *Factory {
+	f.modelConfigService = s
+	return f
+}
```

Comment:
The factory testing pattern should be removed and/or replaced with either mocks or specific integration tests.

Considering we're trying to close out the model config, I'm ok with this, but collective activism should be removing this.